src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -6962,7 +6966,36 @@ static void sofia_handle_sip_r_invite(switch_core_session_t *session, int status
 				}
 			}
 		}
-
+		
+		if(status == 200 && switch_channel_test_flag(channel, CF_RECOVERED) && !tech_pvt->recovered_call_route_fixed && !profile->disable_recovery_record_route_fixup && !switch_true(switch_channel_get_variable(channel, "disable_recovery_record_route_fixup"))) {
```

Comment:
I would suggest that the feature flipper is a OR condition and channel variable should take precedence (which overwrite the sofia profile).

Reply:
The default value for profile->disable_recovery_record_route_fixup and the chanvar disable_recovery_record_route_fixup is 0 for both, meaning that we DO in fact do the fixup.

thus the logic for the flipper is, in pseudo'ish code:    if(  !(0) && !(0) ) { do_fixup; }, which translates to if ( 1 && 1 ) { do_fixup; }

It only takes EITHER profile->disable_recovery_record_route_fixup to be set to 1 OR the chanvar disable_recovery_record_route_fixup to be set to 1 for the entire if() to evaluate to 0, thus the fixup is NOT performed.

I don't see a more effective manner to do this with an || conditional given this.

Reply:
Changed.  Now the feature flipper gives precedence to the channel variable.  It can therefore override whatever is set on the sofia profile now.